File: Course_01/C Language/Vetores/comerciante.c
#include <stdio.h>
#include <stdlib.h>

int main()
{
    int n;

    printf("Serao digitados dados de quantos produtos? ");
    scanf("%d", &n);

    char nome[n][50];
    double precoCompra[n], precoVenda[n], lucro[n], totalCompra, totalVenda, lucroTotal, pLucro[n];
    int count10, count1020, count20;

    for (int i = 0; i < n; i++) {
        printf("Produto %d: \n", i + 1);
        printf("Nome: ");
        fseek(stdin, 0, SEEK_END);
        fgets(&nome[i], 50, stdin);
        printf("Preco de compra: ");
        scanf("%lf", &precoCompra[i]);
        printf("Preco de venda: ");
        scanf("%lf", &precoVenda[i]);
        lucro[i] = precoVenda[i] - precoCompra[i];
        pLucro[i] = (lucro[i] / precoCompra[i]) * 100;
    }

    count10 = 0;
    count1020 = 0;
    count20 = 0;
    totalCompra = 0;
    totalVenda = 0;

    for (int i = 0; i < n; i++) {
        if (pLucro[i] < 10) {
            count10++;
        }
        else if (pLucro[i] > 20) {
            count20++;
        }
        else {
            count1020++;
        }
        totalCompra += precoCompra[i];
        totalVenda += precoVenda[i];
    }

    lucroTotal = totalVenda - totalCompra;

    printf("RELATORIO:\n");
    printf("Lucro abaixo de 10%%: %d\n", count10);
    printf("Lucro entre 10%% e 20%%: %d\n", count1020);
    printf("Lucro acima de 20%%: %d\n", count20);
    printf("Valor total de compra: %.2lf\n", totalCompra);
    printf("Valor total de venda: %.2lf\n", totalVenda);
    printf("Lucro total: %.2lf\n", lucroTotal);

    return 0;
}

File: Course_01/Java Language/Vetores/Comerciante.java
package Vetores;

import java.util.Locale;
import java.util.Scanner;

public class Comerciante {

	public static void main(String[] args) {
		Locale.setDefault(Locale.US);
		Scanner sc = new Scanner (System.in);
		
		int i, n;
		
		System.out.print("Serão digitados dados de quantos produtos? ");
		n = sc.nextInt();
		
		String [] nome = new String [n];
		double [] compra = new double [n];
		double [] venda = new double [n];
		
		for (i = 0; i < n; i++) {
			System.out.println("Produto " + (i + 1));
			System.out.print("Nome: ");
			sc.nextLine();
			nome[i] = sc.nextLine();
			System.out.print("Preço de compra: ");
			compra[i] = sc.nextDouble();
			System.out.print("Preço de venda: ");
			venda[i] = sc.nextDouble();
		}
		
		int conteLucroMenos10, conteLucroMais20;
		double lucro, lucroTotal, compraTotal, vendaTotal, plucro;
		
		lucroTotal = 0;
		compraTotal = 0;
		vendaTotal = 0;
		conteLucroMenos10 = 0;
		conteLucroMais20 = 0;
		
		for (i = 0; i < n; i++) {
			lucro = venda[i] - compra[i];
			plucro = (lucro * 100) / compra[i];
			if (plucro < 10) {
				conteLucroMenos10++;
			}
			else if (plucro > 20) {
				conteLucroMais20++;
			}
			lucroTotal += lucro;
			compraTotal += compra[i];
			vendaTotal += venda[i];
		}
		
		System.out.println();
		System.out.println("RELATÓRIO:");
		System.out.println("Lucro abaixo de 10%: " + conteLucroMenos10);
		System.out.println("Lucro entre 10% e 20%: " + (n - (conteLucroMenos10 + conteLucroMais20)));
		System.out.println("Lucro acima de 20%: " + conteLucroMais20);
		System.out.println("Valor total de compra: " + compraTotal);
		System.out.println("Valor total de venda: " + vendaTotal);
		System.out.println("Lucro Total: " + String.format("%.2f", lucroTotal));
		
		
		sc.close();
	}

}
